modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -1175,6 +1226,9 @@ private String demandRoutineInfo(UUID supplier, GridDhtPartitionSupplyMessage su
          * Before sending messages, method awaits partitions clearing for full partitions.
          */
         public void requestPartitions() {
+            if (nonNull(stat))
```

Comment:
I think it is a bad design: when we are checking something on null, meaning that rebalance statistic enabled or not.